rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -626,23 +626,29 @@ private long setCorrectEndBlock(long endBlock, long fileLength, String remotePat
     return endBlock;
   }
 
+  private void skipInitializeCache()
+  {
+    totalAvailableForCacheInMB = 0;
+    fileInfoCache = CacheBuilder.newBuilder().build(
+            new CacheLoader<String, FileInfo>()
+            {
+              @Override
+              public FileInfo load(String key) throws Exception
+              {
+                throw new UnsupportedOperationException(
+                        String.format("unexpected load call for key %s; cache disabled on master node", key));
```

Comment:
Take this message as arg to function. We can now disable cache on worker too with your changes.